packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/INodeIncrementalLayoutEngineRegistry.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.diagrams.layout.incremental;
+
+import java.util.List;
+
+/**
+ * The registry of all {@link INodeIncrementalLayoutEngine}.
+ *
+ * @author gcoutable
+ */
+public interface INodeIncrementalLayoutEngineRegistry {
```

Comment:
Why do you need to have this registry? instead of injecting a `List<INodeIncrementalLayoutEngine>` where it's relevant?